deploy/operator/setup_assisted_operator.sh

```diff
@@ -211,6 +211,21 @@ EOCR
   echo "Enabling configuration of BMH resources outside of openshift-machine-api namespace"
   oc patch provisioning provisioning-configuration --type merge -p '{"spec":{"watchAllNamespaces": true}}'
 
+  # Patching the Provisioning CR triggers a rollout of the metal3 deployment.
+  # Since the metal3 deployment has just 1 replica, during the rollout the
+  # metal admission webhook is unavailable. As a result any interaction during
+  # that time with BMH resources causes an error (as the webhook is down).
+  #
+  # This is why before proceeding we have to wait for that process to complete.
+  # This includes waiting a few seconds for the baremetal operator to notice
+  # our patch (there's no simple robust way to do this. It usually happens
+  # almost immediately, but we wait 10 seconds just in case). Then we wait for
+  # the deployment to finish rolling out, and finally we wait for the metal3
+  # pod to be ready.
+  sleep 10 # Wait for the operator to notice our patch
+  oc rollout status -n openshift-machine-api deployment/metal3
```

Comment:
Can we add some timeout for this action?
I guess it should at least finish in let's say a couple of minutes at most

It's both for the user experience of running this test, for having shorter times in cases of errors, and so that we'll fast-fail and have artifacts truncated with the exact logs responsible for the problem

Reply:
Good point